paparazzi/paparazzi/src/main/java/app/cash/paparazzi/accessibility/A11ySnapshotHandler.kt

```diff
@@ -0,0 +1,217 @@
+    }
+
+    private fun Graphics2D.withComposite(newComposite: Composite, fn: () -> Unit = {}) {
+      val current = composite
+      composite = newComposite
+
+      fn()
+
+      composite = current
+    }
+
+    private fun BufferedImage.withGraphics2D(fn: Graphics2D.() -> Unit = {}): BufferedImage {
+      createGraphics().apply {
+        try {
+          fn()
+        } finally {
+          dispose()
+        }
+      }
+
+      return this
+    }
+
+    internal fun drawBoxes(
+      accessibilityState: AccessibilityState,
+      image: BufferedImage
+    ): BufferedImage {
+      val modifiedImage = BufferedImage(image.width, image.height, image.type)
+
+      val scale = 1000f / max(accessibilityState.height, accessibilityState.width)
+
+      return modifiedImage.withGraphics2D {
+        withComposite(AlphaComposite.getInstance(AlphaComposite.SRC_OVER, 0.5f)) {
+          drawImage(image, 0, 0, image.width, image.height, null)
+        }
+
+        accessibilityState.elements.forEachIndexed { i, it ->
+          paint = colorForIndex(i)
+          stroke = BasicStroke(3f)
+          val element = it.scaleBy(scale)
+          drawRect(
+            element.displayBounds.left,
+            element.displayBounds.top,
+            element.displayBounds.width(),
+            element.displayBounds.height()
+          )
+          paint = Color(color.red, color.green, color.blue, 255 / 3)
+          fillRect(
+            element.displayBounds.left,
+            element.displayBounds.top,
+            element.displayBounds.width(),
+            element.displayBounds.height()
+          )
+          if (element.touchBounds != null) {
+            drawRect(
+              element.touchBounds.left,
+              element.touchBounds.top,
+              element.touchBounds.width(),
+              element.touchBounds.height()
+            )
+          }
+        }
+      }
+    }
+
+    internal fun drawLegend(
+      accessibilityState: AccessibilityState,
+      image: BufferedImage
+    ): BufferedImage {
+      val modifiedImage = BufferedImage(600, image.height, image.type)
+
+      return modifiedImage.withGraphics2D {
+        paint = Color.WHITE
+        fillRect(0, 0, modifiedImage.width, modifiedImage.height)
+
+        font = font.deriveFont(20f)
+        stroke = BasicStroke(3f)
+
+        var index = 1
+        accessibilityState.elements.forEachIndexed { i, it ->
+          paint = Color.BLACK
+
+          val start = index
+          if (it.role != null || it.disabled) {
+            val role = if (it.role != null) "Role " + it.role + " " else ""
+            val heading = if (it.heading) "Heading " else ""
+            val disabled = if (it.disabled) "Disabled" else ""
+            drawString(role + heading + disabled, 50f, 28f * index++)
+          }
+          if (it.contentDescription != null) {
+            drawString(
+              "Content Description " + it.contentDescription.joinToString(", "),
+              1050f,
```

Comment:
This should be 50f.

Reply:
Why? It will draw over the image in that case.

Reply:
The image is created with width = 600f: https://github.com/cashapp/paparazzi/pull/555/files/85eaa50bce25715fa7d4c41eac6c870ba74fc067#diff-d564e3b448efd8866f9bbbf1ea2d0a900d0892e6a38c5d73785218a3906ddd2eR165 and all other calls use 50f for the x axis, for example: https://github.com/cashapp/paparazzi/pull/555/files/85eaa50bce25715fa7d4c41eac6c870ba74fc067#diff-d564e3b448efd8866f9bbbf1ea2d0a900d0892e6a38c5d73785218a3906ddd2eR196

With 1050f we'll be drawing outside the image.